toolkit/tools/pkg/imagecustomizerlib/customizepartitions_test.go

```diff
@@ -97,4 +100,38 @@ func TestCustomizeImagePartitions(t *testing.T) {
 		expectedSource := fmt.Sprintf("PARTUUID=%s", partition.PartUuid)
 		assert.Equalf(t, expectedSource, fstabEntry.Source, "fstab [%d]: source", i)
 	}
+
+	// Check that the extraCommandLine was added to the grub.cfg file.
+	grubCfgContents, err := file.Read(grubCfgFilePath)
+	assert.NoError(t, err, "read grub.cfg file")
+	assert.Regexp(t, "linux.* console=tty0 console=ttyS0 ", grubCfgContents)
+}
+
+func TestCustomizeImageKernelCommandLine(t *testing.T) {
+	var err error
+
+	baseImage := checkSkipForCustomizeImage(t, baseImageTypeCoreEfi)
+
+	buildDir := filepath.Join(tmpDir, "TestCustomizeImageCopyFiles")
+	configFile := filepath.Join(testDir, "extracommandline-config.yaml")
+	outImageFilePath := filepath.Join(buildDir, "image.qcow2")
+
+	// Customize image.
+	err = CustomizeImageWithConfigFile(buildDir, configFile, baseImage, nil, outImageFilePath, "raw", "",
+		false /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	imageConnection, err := connectToCoreEfiImage(buildDir, outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer imageConnection.Close()
+
+	// Check that the extraCommandLine was added to the grub.cfg file.
+	grubCfgFilePath := filepath.Join(imageConnection.Chroot().RootDir(), "/boot/grub2/grub.cfg")
+	grubCfgContents, err := file.Read(grubCfgFilePath)
+	assert.NoError(t, err, "read grub.cfg file")
+	assert.Regexp(t, "linux.* console=tty0 console=ttyS0 ", grubCfgContents)
```

Comment:
is it better to use different kernel args to validate this? I don't recall if the console args are also present in the efi core image or not, I know for the base images they are there by default.

Reply:
This test explicitly takes the core-efi image as input. And that image doesn't contain this args.